src/net.cpp

```diff
@@ -2865,3 +2867,25 @@ uint64_t CConnman::CalculateKeyedNetGroup(const CAddress& ad) const
 
     return GetDeterministicRandomizer(RANDOMIZER_ID_NETGROUP).Write(vchNetGroup.data(), vchNetGroup.size()).Finalize();
 }
+
+void LogMessage(CNode* node, std::string& command, std::vector<unsigned char>& data, bool is_incoming)
+{
+    int64_t time = GetTimeMicros();
```

Comment:
Nit: Could be introduced when needed?

Rationale: [ES.21: Don’t introduce a variable (or constant) before you need to use it](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-introduce) (C++ Core Guidelines) :)

Reply:
I'd prefer to keep this as it is.  I understand (and agree with) ES.21 completely, but I'm trying to treat `time` as an argument, not a local, which is why it's right at the top.  I would love to pass it in instead but practically can't (see [here](https://github.com/bitcoin/bitcoin/pull/19509#discussion_r458760191)).

If anyone is strongly against this I'll move it.